### PR TITLE
Access syntax using index.

### PR DESCRIPTION
**Pull request by noppefoxwolf**
Fixed iterator access using index as follows line.
It can be efficient allocate memory and access syntax using `lazy` and `prefix`.

```
let functionCallExpressionMap = Array(functionCallExpression.children.makeIterator().lazy.map({ $0 }).prefix(3))
```

### CODE REVIEW

File: Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift [hunk 1]
```diff
@@ -20,40 +20,38 @@ class TranslateTransformer: SyntaxRewriter {
     }
 
     override func visit(_ functionCallExpression: FunctionCallExprSyntax) -> ExprSyntax {
-        var functionCallExpressionIterator = functionCallExpression.children.makeIterator()
+        let functionCallExpressionMap = Array(functionCallExpression.children.makeIterator().lazy.map({ $0 }).prefix(3))
```
Comment (Jeehut) [inline Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift:23]: Why not simply run `Array(functionCallExpression.children).prefix(3)`?

Comment (noppefoxwolf) [reply Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift:23]: if `functionCallExpression` has n children (n > 2). Swift allocate memory for n * struct.
But this code allocate 3 * struct.
It's more efficient than allocate full range array.

Comment (Jeehut) [reply Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift:23]: I see, thank you for the explanation, makes sense.

Comment (Jeehut) [reply Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift:23]: Why is the `.map { $0 }` needed though?

Comment (noppefoxwolf) [reply Sources/BartyCrouchKit/FileHandling/TranslateTransformer.swift:23]: It's sorry. not needed.
I'll fix soon. 😅

**Comment by Jeehut** [comment]
@noppefoxwolf Thank you very much for this subsequent PR, but you were a little too fast as I was still working on my PR, mainly fixing SwiftLint warnings to get the CI going. Now we have some merge conflicts here (I've changed the target branch to `main` because I already merged my PR), could you please fix them? I think the main difference is the `.makeIterator()` call which I didn't put in there. Can you check if everything else is correct? I copied the code manually during the process before you posted this PR and then tried to adjust.

Feel free to also close this and start a new PR if you find that easier. Sorry for the inconvenience.

ドイツからのご挨拶！

**Comment by Jeehut** [comment]
@noppefoxwolf I have just diffed the code from your branch and the `main` branch and applied any changes that were missing in c74ea0d860feca5971355609639c2d75a58a7685. Closint this then, thank you very much for your efforts, will release a new version with this fix now. 👍

**Pull request closed without merging**